packages/mobile/capacitor/plugins/secure-filesystem-access/android/src/main/java/org/iota/plugins/securefilesystemaccess/SecureFilesystemAccessPlugin.java

```diff
@@ -0,0 +1,325 @@
+        if (!call.getData().has("oldName")
+            || !call.getData().has("newName")) {
+            call.reject("newName is required");
+            return;
+        }
+        String oldName = call.getString("oldName");
+        String newName = call.getString("newName");
+        assert oldName != null;
+        File oldFilename = new File(getContext().getFilesDir(), oldName);
+        assert newName != null;
+        File newFilename = new File(getContext().getFilesDir(), newName);
+        boolean isRenamed = false;
+        String error = "";
+        try {
+            isRenamed = oldFilename.renameTo(newFilename);
+        } catch (Exception e) {
+            error = Objects.requireNonNull(e.getCause()).toString();
+        }
+        if (isRenamed) {
+            call.resolve();
+        } else {
+            call.reject(error);
+        }
+    }
+
+    @PluginMethod
+    public void listProfileFolders(PluginCall call) {
+        if (!call.getData().has("folder")) {
+            call.reject("folder is required");
+            return;
+        }
+        String folder = call.getString("folder");
+        String[] files = null;
+        File fileObject = new File(getContext().getFilesDir(), "__storage__" + File.separator + folder);
+        if (fileObject.exists()) {
+            files = fileObject.list();
+        } else {
+            call.reject("Folder does not exist");
+        }
+        JSObject response = new JSObject();
+        response.put("folderList", files);
+        call.resolve(response);
+    }
+
+    @PluginMethod
+    public void showPicker(PluginCall call) {
+        if (getPermissionState("storage") != PermissionState.GRANTED) {
+            requestPermissionForAlias("storage", call, "pickerPermsCallback");
+        } else {
+            if (!call.getData().has("type")) {
+                call.reject("Resource type is required");
+                return;
+            }
+            resourceType = call.getString("type");
+            assert resourceType != null;
+
+            Intent intent = new Intent(resourceType.equals("file")
+                ? Intent.ACTION_OPEN_DOCUMENT : Intent.ACTION_OPEN_DOCUMENT_TREE);
+            intent.addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION);
+
+            if (resourceType.equals("file")) {
+                intent.setType("*/*");
+                intent.addCategory(Intent.CATEGORY_OPENABLE);
+                intent.putExtra(Intent.EXTRA_ALLOW_MULTIPLE, false);
+
+            } else if (resourceType.equals("folder")) {
+                intent.putExtra(Intent.EXTRA_LOCAL_ONLY, true);
+                intent.addFlags(Intent.FLAG_GRANT_WRITE_URI_PERMISSION);
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+                    // this is not working on Android 11+
+                    intent.putExtra(DocumentsContract.EXTRA_INITIAL_URI,
+                            Uri.parse(Environment.getExternalStoragePublicDirectory(DIRECTORY_DOWNLOADS).toString()));
```

Comment:
Thanks, replaced!

Reply:
I've tested with getExternalFilesDir() but it seems that this function return us a really deep path that is also deleted when the app is uninstalled, and where an average user could not find imo.
We would need Environment.getExternalStoragePublicDirectory() even deprecated since we need that Stronghold have permissions to write the backup file on an public accessible folder. Also I could not find a better option that works with Capacitor.